core/src/main/java/io/parsingdata/metal/expression/value/reference/Ref.java

```diff
@@ -50,7 +51,7 @@
  * the amount of returned results.
  * @param <T> The type of reference to match on.
  */
-public class Ref<T> implements ValueExpression {
+public abstract class Ref<T> implements ValueExpression {
```

Comment:
Did we just forget to mark this `abstract`? I try to remember if we had concrete plans to actually use this class in a stand-alone fashion...

Reply:
Good point! I think the abstract modifier could have been added earlier. Since we added an abstract method in this PR, it is mandatory now.